packages/cadl-ranch-specs/http/client/naming/main.tsp

```diff
@@ -187,3 +187,53 @@ namespace Model {
   @post
   op language(...ModelWithLanguageClientName): NoContentResponse;
 }
+
+@operationGroup
+@route("/enum")
+namespace Enum {
+  @clientName("ClientExtensibleEnum")
+  union ServerExtensibleEnum {
+    string,
+    EnumValue1: "value1",
+  }
+
+  union ExtensibleEnum {
+    string,
+
+    @clientName("ClientEnumValue1")
+    EnumValue1: "value1",
+
+    @clientName("ClientEnumValue2")
+    "value2",
+  }
+
+  @scenario
+  @scenarioDoc("""
+    Testing that we can project a enum name and enum value name.
+    Your generated SDK should generate an Enum "ClientExtensibleEnum".
+    (The exact name may depend on language convention)
+
+    Expected request body:
+    ```json
+    "value1"
+    ```
+  """)
+  @route("/enum-name")
+  @post
+  op enumName(@body body: ServerExtensibleEnum): NoContentResponse;
+
+  @scenario
+  @scenarioDoc("""
+    Testing that we can project a enum name and enum value name.
+    Your generated SDK should generate an Enum with members "ClientEnumValue1", "ClientEnumValue2".
+    (The exact name may depend on language convention)
+
+    Expected request body:
+    ```json
+    "value1"
+    ```
+  """)
+  @route("/enum-member-name")
+  @post
+  op enumMemberName(@body body: ExtensibleEnum): NoContentResponse;
```

Comment:
i may not get the idea of enum-member-name here because i notice the body type is union(finally enum) not enum member?

Reply:
oh this is for client name testing, i just image the case below:
```
op enumMemberName(@body body: ExtensibleEnum.EnumValue1): NoContentResponse;
```

Reply:
I could call it e.g. `union-enum-member-name` is that's sound better to you :-)

Personally I don't have good idea to say in the method name that "this type is union but it should be treated as enum..."

Reply:
haha, interesting i am wondering if we can add @clientName for real tsp enum?

Reply:
Sure. I've added prefix `union` to the case, so that we can use `enum` in future when do fixed enum.